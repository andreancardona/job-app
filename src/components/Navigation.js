import React from 'react';
import profilePlaceholder from '../images/profilePlaceholder.svg';
import '../styles/Navigation.css';

const Navigation = () => {
  return (
    <div className="nav-container">
      <div className="nav-title" tabIndex="0" >Welcome to JobHob</div>
        <div className="user-profile">
          <img className="user-profile--image" alt="profile placeholder" tabIndex="0"  src={profilePlaceholder} />
          <div className="user-profile--info">
            <div className="user-profile--name" tabIndex="0">Andrea Cardona</div>
            <div className="user-profile--role" tabIndex="0">Admin</div>
          </div>
        </div>
     </div>
  );
}

export default Navigation;