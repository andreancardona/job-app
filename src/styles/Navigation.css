/* banner */

.banner {
  position: absolute;
  height: 64px;
  left: 0px;
  top: 0px;
  background: #2D2D2D;
  width: 100%;
}
 
 /* navigation */

 .nav-container {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  width: 100%;
 }

.header {
  color: white;
  font-family: Helvetica Neue;
  margin: 20px 20px 0 20px;
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  letter-spacing: 0em;
  height: 10px;
}

/* user profile */

.user-profile--image {
  width: 36px;
  height: 36px;
  left: 822px;
  margin: 12px 8px 0 0;
}

.user-info {
  display: flex;
  flex-direction: column;
  margin-top: 11px;
}

.user-profile--name {
  left: 866px;
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 150%;
  color: #FAF9F8;
}

.user-profile--role {
  width: 35px;
  height: 18px;
  left: 866px;
  top: 35px;
  font-family: Helvetica Neue;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 150%;
  color: #FAF9F8;
}