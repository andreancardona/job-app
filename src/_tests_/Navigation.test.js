import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import Navigation from '../components/Navigation';

 // Navigation props
 const navProps = {
  title: 'JobHob',
  userName: 'Andrea Cardona',
  userRole: 'Admin'
}

describe('The <Navigation /> component', () => {
  test('Navigation renders with Navigation props', () => {
    const tree = renderer.create(<Navigation props={navProps}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('Job should not fail any accessibility tests', async () => {
    const { container } = render(<Navigation props={navProps}/>);
    expect(await axe(container)).toHaveNoViolations();
  });
});
