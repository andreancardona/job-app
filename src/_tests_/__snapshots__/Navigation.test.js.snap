// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Navigation /> component Navigation renders with Navigation props 1`] = `
<nav
  className="nav-container"
  id="nav"
>
  <div
    className="nav-title"
  >
    JobHob
  </div>
  <a
    className="skip-content"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="user-profile"
  >
    <img
      alt="profile placeholder"
      className="user-profile--image"
      src="profilePlaceholder.svg"
    />
    <div
      className="user-profile--info"
    >
      <div
        className="user-profile--name"
      >
        Andrea Cardona
      </div>
      <div
        className="user-profile--role"
      >
        Admin
      </div>
    </div>
  </div>
</nav>
`;
